Feature dataapi 5 return null for non string values

Pixl8RusselCole opened the pull request.

Pixl8RusselCole committed:
[twgit] Init feature 'feature-DATAAPI-5_Return_NULL_for_non_string_values'.

Pixl8RusselCole committed:
DATAAPI-5 NULL for numerics and feature flag to use quotes instead

DominicWatson commented on config/Config.cfc:
```diff
@@ -16,6 +16,7 @@ component {
 		settings.features.restTokenAuth.enabled = true;
 
 		settings.features.dataApiQueue = settings.features.dataApiQueue ?: { enabled=true };
+		settings.features.dataApiQuotesForNumerics = settings.features.dataApiQuotesForNumerics ?: { enabled=false };
```
I think this feature name is a little unclear. Could we use `dataApiUseNullForNumerics` and set to _enabled_ by default instead? (and then update the logic that uses it)

DominicWatson commented on services/DataApiService.cfc:
```diff
@@ -303,12 +303,18 @@ component {
 
 	private any function _renderField( required any value, required string renderer, struct fieldSettings={} ) {
+
+		var emptyNumericQuotes = false;
```
I think we should be able to ditch these four lines (which could have been just one line anyway - `var emptyNumericQuotes = $isFeatureEnabled( "..." );`).

Just use the `$isFeatureEnabled()` call _only_ when dealing with the numeric type:

```cfc
case "numeric": 
    if ( IsNumeric( arguments.value ) ) {
        return arguments.value;
    }
    return $isFeatureEnabled( "..." ) ? NullValue() ?: "";
```

I also just spotted that this is now returning `true` or `false` instead of the number!

DominicWatson reviewed (changes requested):
Nice one, looking good. A couple of bits of feedback on this though. See review comments.

Pixl8RusselCole committed:
DATAAPI-5
- change dataApiQuotesForNumerics.enabled=false to dataApiUseNullForNumerics.enabled=true
- rework _renderField numeric
